services/task-svc/internal/repositories/patient_repository.go

```diff
@@ -109,6 +109,22 @@ func (r *PatientRepository) GetRoomsWithBedsWithActivePatientsForOrganization(or
 	return rooms, nil
 }
 
+func (r *PatientRepository) GetRoomsWithBedsWithActivePatientsForWard(wardID uuid.UUID) ([]models.Room, error) {
+	var rooms []models.Room
+	query := r.db.
+		Preload("Beds.Patient").
+		Joins("JOIN beds ON rooms.id = beds.room_id").
+		Joins("JOIN patients ON patients.bed_id = beds.id").
+		Where("rooms.ward_id = ? AND patients.is_discharged = 0", wardID).
+		Group("rooms.id").
```

Comment:
I'm confused about the group here. We `.Find()` on `models.Room`s, so we should get rooms already, right?
Is a `LEFT` or `INNER JOIN` sufficient here?

Reply:
I just altered the code to match the ward scope, but I will open an issue to look into these join queries